Many string truncation warnings with newer GCC

**Describe the bug**
Ubuntu has released 20.04 LTS which includes GCC 9.3.0.  When using this compiler it implements a much stricter (and often over-zealous) checking of string ops.

For instance: 
```
In file included from /usr/include/string.h:495,
                 from /home/joe/code/cfecfs/github/psp/fsw/pc-linux/src/cfe_psp_start.c:37:
In function ‘strncpy’,
    inlined from ‘CFE_PSP_ProcessArgumentDefaults’ at /home/joe/code/cfecfs/github/psp/fsw/pc-linux/src/cfe_psp_start.c:524:7:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:106:10: error: ‘__builtin_strncpy’ specified bound 32 equals destination size [-Werror=stringop-truncation]
  106 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**To Reproduce**
Build code with default config using GCC 9.3.0, with optimization enabled and full warnings.

**Expected behavior**
Code should build cleanly.

**System observed on:**
Ubuntu 20.04 LTS 64 bit.

**Reporter Info**
Joseph Hickey, Vantage Systems, Inc.

--- a/fsw/pc-linux/src/cfe_psp_start.c
+++ b/fsw/pc-linux/src/cfe_psp_start.c
@@ -186,7 +186,9 @@ void OS_Application_Startup(void)
       switch( opt ) 
       {
          case 'R':
-            strncpy(CommandData.ResetType, optarg, CFE_PSP_RESET_NAME_LENGTH);
+            strncpy(CommandData.ResetType, optarg, CFE_PSP_RESET_NAME_LENGTH-1);
+            CommandData.ResetType[CFE_PSP_RESET_NAME_LENGTH-1] = 0;
+
             if ((strncmp(CommandData.ResetType, "PO", CFE_PSP_RESET_NAME_LENGTH ) != 0 ) &&
                 (strncmp(CommandData.ResetType, "PR", CFE_PSP_RESET_NAME_LENGTH ) != 0 ))
             {
@@ -214,7 +216,8 @@ void OS_Application_Startup(void)
             break;
 
          case 'N':
-            strncpy(CommandData.CpuName, optarg, CFE_PSP_CPU_NAME_LENGTH );
+            strncpy(CommandData.CpuName, optarg, CFE_PSP_CPU_NAME_LENGTH-1 );
+            CommandData.CpuName[CFE_PSP_CPU_NAME_LENGTH-1] = 0;
             printf("CFE_PSP: CPU Name: %s\n",CommandData.CpuName);
             CommandData.GotCpuName = 1;
             break;
@@ -521,7 +524,8 @@ void CFE_PSP_ProcessArgumentDefaults(CFE_PSP_CommandData_t *CommandDataDefault)
    
    if ( CommandDataDefault->GotCpuName == 0 )
    {
-      strncpy(CommandDataDefault->CpuName, CFE_PSP_CPU_NAME, CFE_PSP_CPU_NAME_LENGTH );
+      strncpy(CommandDataDefault->CpuName, CFE_PSP_CPU_NAME, CFE_PSP_CPU_NAME_LENGTH-1 );
+      CommandDataDefault->CpuName[CFE_PSP_CPU_NAME_LENGTH-1] = 0;
       printf("CFE_PSP: Default CPU Name: %s\n",CFE_PSP_CPU_NAME);
       CommandDataDefault->GotCpuName = 1;
    }

